frontend/src/ui/NotificationContext.tsx

```diff
@@ -0,0 +1,56 @@
+import { bug, Card } from "@opencast/appkit";
+import { createContext, useState, useContext, PropsWithChildren, useEffect, useMemo } from "react";
+import { useRouter } from "../router";
+
+export type NotificationMessage = {
+    kind: "info" | "error";
+    // Making this a function helps the message to use the currently
+    // selected language when changed.
+    message: () => string;
```

Comment:
Mhhh I think it would be cleaner if that would take `i18n` as argument. The reason it works right now is that by lucky coincidence, the component that renders this message and thus calls the function, is rerendered on language change _due to something else_. Very likely it is just using `useTranslation` for some other string. And fair, most components will do that anyway. But a component that does not use `useTranslation` will not get rerendered on language change and thus this method will not be called again, but the old string is used. Requiring `i18n` forces the calling component to use `useTranslation`.